api/proto/teleport/userpreferences/v1/userpreferences.proto

```diff
@@ -51,6 +52,8 @@ message UpsertUserPreferencesRequest {
   UserPreferences preferences = 1;
   // username is the username of the owner of the user preferences to update.
   string username = 2;
+  // fieldmask is the field mask of the user preferences to update.
+  google.protobuf.FieldMask fieldmask = 3;
```

Comment:
At the moment it's the full path, i.e. `assist.preferred_logins`. This is because the frontend code doesn't check if you're updating all of the object's values to simplify `assist.view_mode` and `assist.preferred_logins` into `assist`.

Do you think I should add support for specifying `assist`?

Also, I updated all the names to update mask.

Reply:
There are no hard-and-fast rules for this, I think. Just document your decision and it's alright.